File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/BabyNames/BabyNames_CSharp/Queries.cs
//--------------------------------------------------------------------------
// 
//  Copyright (c) Microsoft Corporation.  All rights reserved. 
// 
//  File: Graph.cs
//
//--------------------------------------------------------------------------

using System;
using System.Linq;
using System.Windows;

namespace BabyNames
{
    public partial class Window1 : Window
    {
        private void InitializeQueries()
        {
            int numProcs = (int)slProcessorsToUse.Value;

            // SEQUENTIAL QUERY
            _sequentialQuery = from b in _babies
                               where b.Name.Equals(_userQuery.Name, StringComparison.InvariantCultureIgnoreCase) &&
                                     b.State == _userQuery.State &&
                                     b.Year >= YEAR_START && b.Year <= YEAR_END
                               orderby b.Year
                               select b;

            // PARALLEL QUERY
            _parallelQuery = from b in _babies.AsParallel().WithDegreeOfParallelism(numProcs)
                             where b.Name.Equals(_userQuery.Name, StringComparison.InvariantCultureIgnoreCase) &&
                                   b.State == _userQuery.State &&
                                   b.Year >= YEAR_START && b.Year <= YEAR_END
                             orderby b.Year
                             select b;
        }
    }
}

File: Samples for Parallel Programming with the .NET Framework/[C#][VB][C++][F#]-Samples for Parallel Programming with the .NET Framework/C# and C++ and F# and VB/C#,C++,F#,VB/BabyNames/BabyNames_VisualBasic/Queries.vb
'--------------------------------------------------------------------------
' 
'  Copyright (c) Microsoft Corporation.  All rights reserved. 
' 
'  File: Graph.vb
'
'--------------------------------------------------------------------------

Namespace BabyNames
	Partial Public Class Window1
		Inherits Window
		Private Sub InitializeQueries()
            Dim numProcs = CInt(Fix(slProcessorsToUse.Value))

			' SEQUENTIAL QUERY
            _sequentialQuery = From b In _babies
                               Where b.Name.Equals(_userQuery.Name, StringComparison.InvariantCultureIgnoreCase) AndAlso
                                     b.State = _userQuery.State AndAlso
                                     b.Year >= YEAR_START AndAlso b.Year <= YEAR_END
                               Order By b.Year
                               Select b

			' PARALLEL QUERY
            _parallelQuery = From b In _babies.AsParallel().WithDegreeOfParallelism(numProcs)
                             Where b.Name.Equals(_userQuery.Name, StringComparison.InvariantCultureIgnoreCase) AndAlso
                                   b.State = _userQuery.State AndAlso
                                   b.Year >= YEAR_START AndAlso b.Year <= YEAR_END
                             Order By b.Year
                             Select b
		End Sub
	End Class
End Namespace